core/dnsserver/config.go

```diff
@@ -50,6 +52,22 @@ type Config struct {
 	registry map[string]plugin.Handler
 }
 
+//HostAddresses builds a representation of the addresses of this Config
+//after server is started ONLY, can be used as a Key for identifing that config
+// :53 or 127.0.0.1:53 or 127.0.0.1:53/::1:53
+func (c *Config) HostAddresses() string {
+	hosts := ""
+	for _, h := range c.ListenHosts {
+		host := net.JoinHostPort(h, c.Port)
```

Comment:
JoinHostPort doesn't return an empty string

Reply:
I agree, but not sure why you are doing that remark.
Is it because I test if "hosts" is not empty on the next line ? 
=> mistmatch between host and hosts ?